lms/djangoapps/courseware/courses.py

```diff
@@ -129,8 +129,16 @@ def get_course_overview_with_access(user, action, course_key, check_if_enrolled=
     try:
         course_overview = CourseOverview.get_from_id(course_key)
     except CourseOverview.DoesNotExist:
+        log.exception(f'Failed to retrieve course from courseoverview."{course_key}"')
         raise Http404("Course not found.")  # lint-amnesty, pylint: disable=raise-missing-from
+
+    course_under_investigation = str(course_key) == 'course-v1:UQx+ABLE301x+1T2022'
+    if course_under_investigation:
+        log.info('[TNL_9420] Course overview found, Checking course access.')
```

Comment:
Nit: add a TODO here that this needs to be removed later?

Reply:
Yes, adding it now.